test/SourceKit/InterfaceGen/gen_swift_module.swift

```diff
@@ -33,3 +33,51 @@ func f(s : inout [Int]) {
 // RUN: %swift -emit-module -o /dev/null -emit-module-interface-path %t.mod/swift_mod.swiftinterface -O %S/Inputs/swift_mod.swift -parse-as-library -disable-implicit-concurrency-module-import -disable-implicit-string-processing-module-import
 // RUN: %sourcekitd-test -req=interface-gen -module swift_mod -- -Xfrontend -disable-implicit-concurrency-module-import -Xfrontend -disable-implicit-string-processing-module-import  -I %t.mod -module-cache-path %t.mod/mcp > %t.response
 // RUN: %diff -u %s.from_swiftinterface.response %t.response
+
+// Separately test whether the optional declarations array contains the expected information from Inputs/swift_mod.swift
+// RUN: %sourcekitd-test -req=interface-gen -req-opts=enabledeclarations=true -module swift_mod -- -Xfrontend -disable-implicit-concurrency-module-import -Xfrontend -disable-implicit-string-processing-module-import  -I %t.mod -module-cache-path %t.mod/mcp | %FileCheck -check-prefix=CHECK-DECLARATIONS-ARRAY %s
+// check that the other arrays are still there as normal, by checking for a unique identifying member in each
+// CHECK-DECLARATIONS-ARRAY-DAG: key.kind:{{.*}}ref.associatedtype
+// CHECK-DECLARATIONS-ARRAY-DAG: key.kind:{{.*}}syntaxtype
+// CHECK-DECLARATIONS-ARRAY-DAG: key.substructure
+
+// CHECK-DECLARATIONS-ARRAY: {{^}}]{{$}}
+
+// public class MyClass
+// CHECK-DECLARATIONS-ARRAY: key.kind:{{.*}}class
+// CHECK-DECLARATIONS-ARRAY-NEXT: key.usr
+// CHECK-DECLARATIONS-ARRAY-SAME: swift_mod
+// CHECK-DECLARATIONS-ARRAY-SAME: MyClass
+
+// public func pub_method
+// CHECK-DECLARATIONS-ARRAY: key.kind:{{.*}}method.instance
+// CHECK-DECLARATIONS-ARRAY-NEXT: key.usr
+// CHECK-DECLARATIONS-ARRAY-SAME: swift_mod
+// CHECK-DECLARATIONS-ARRAY-SAME: MyClass
+// CHECK-DECLARATIONS-ARRAY-SAME: pub_method
+
+// CHECK-DECLARATIONS-ARRAY-NOT: int_method
+// CHECK-DECLARATIONS-ARRAY-NOT: fp_method
+// CHECK-DECLARATIONS-ARRAY-NOT: priv_method
+
+// public protocol MyProto
+// CHECK-DECLARATIONS-ARRAY: key.kind:{{.*}}protocol
+// CHECK-DECLARATIONS-ARRAY-NEXT: key.usr
+// CHECK-DECLARATIONS-ARRAY-SAME: swift_mod
+// CHECK-DECLARATIONS-ARRAY-SAME: MyProto
```

Comment:
Done, thank you! I didn't check this originally, because the array builder guarantees that this field exists. But it's a good idea to check it nonetheless be a) that makes it harder to change this fact in the array builder without noticing that it breaks things, and b) the key might also be there, but be zero (the default value), which shouldn't be the case. So the updated test also checks that nothing has caused the offsets to be zero.